add async trigger for `hubspot` (#72)
ballerina-platform/asyncapi-triggers: asyncapi/hubspot/Module.md

## Overview

> The [Ballerina](https://ballerina.io/) listener for Hubspot provides capability to listen following events happening in HubSpot account.

Author: Sure. I will get it reviewed

> The [Ballerina](https://ballerina.io/) listener for Hubspot provides capability to listen following events happening in HubSpot account.

Reviewer: ```suggestion
The [Ballerina](https://ballerina.io/) listener for Hubspot allows you to listen to the following events that are carried out in a HubSpot account.
```

Author: Changed as suggested.

* `onCompanyCreation`
* `onCompanyDeletion`
* `onContactCreation`
* `onContactDeletion`
* `onConversationCreation`
* `onConversationDeletion`
* `onDealCreation`
* `onDealDeletion`

## Prerequisites

Before using this connector in your Ballerina application, complete the following:

> - Create [Hubspot account](https://app.hubspot.com/signup-hubspot/)

Reviewer: Isn't the application name formatted `HubSpot`? Format it the same the application owners format it. This comment applies to all the places where it is used.

> - Create [Hubspot account](https://app.hubspot.com/signup-hubspot/)

Author: Modified the changes.

- Create an [app](https://developers.hubspot.com/get-started) in devoloper portal

> - Go to webhooks and register a targetURL for notifications.

Author: Added

> - Create a subscription with interested events.

Author: Yes. Updated as suggested.

## Quickstart

> To use the Hubspot listener in your Ballerina application, update the .bal file as follows:

Author: Changed

### Step 1: Import listener

> Import the ballerinax/trigger.hubspot module into the Ballerina project.

Reviewer: ```suggestion
To import the `ballerinax/trigger.hubspot` module into the Ballerina project, add the following statement:
```

```ballerina
    import ballerinax/trigger.hubspot;
```

### Step 2: Create a new listener instance

> Create a `listenerConfig` with the AppSecret and CallbackURL. 

Reviewer: ```suggestion
To create a `listenerConfig` with the application secret and the callback URL, add the following configuration.
```
If "configuration" is the wrong word to use here, replace it with a more appropriate word.

```ballerina
    listener hubspot:Listener hubspotWebhook = new(listenerConfig = {clientSecret, callbackURL});
```
**Note:** `ClientSecret` can be obtained from the Hubspot portal.

### Step 3: Invoke listener triggers

> 1. Now you can use the triggers available within the listener. 

Reviewer: ```suggestion
Now let's use the triggers available within the listener.
```

>     Following is an example of how to listen for events on Company creation and deletion.

Reviewer: ```suggestion
For example, you can configure the Ballerina listener to listen to company creation and deletion events as follows:
```

>     Listen to Hubspot Company Creation and Deletion

Reviewer: Is this a comment within the `.bal` file? If not, shall we leave it out because it causes a repetition?

    ```ballerina
        service hubspot:CompanyService on hubspotWebhook {
            remote function onCompanyCreation(hubspot:WebhookEvent event) returns error? {
                io:println(event);
            }

            remote function onCompanyDeletion(hubspot:WebhookEvent event) returns error? {
                io:println(event);
            }
        }
    ```

>     **Note:** Event payload doesn't contain metadata related to the event. You need use the specific Hubspot client to obtain it.

Reviewer: ```suggestion
**Note:** The event payload does not contain metadata related to the event. You need use the specific HubSpot client to obtain it.
```

Author: Changed

> 2. Use `bal run` command to compile and run the Ballerina program. 

Reviewer: ```suggestion
To compile and run the Ballerina program, issue the following command.

`bal run`
```

Author: Changed as suggested.
